File: stress/t/arr_dotprod.f
      PROGRAM ARRDOTPROD
      INTEGER*8 I,N,A,B,DOT,T
      INTEGER*8 U(0:31),W(0:31)
      N=32
      I=0
   10 IF (I.GE.N) GOTO 20
      U(I)=I-16
      W(I)=2*I+3
      I=I+1
      GOTO 10
   20 CONTINUE
      DOT=0
      I=0
   30 IF (I.GE.N) GOTO 40
      A=U(I)
      B=W(I)
      DOT=DOT+A*B
      I=I+1
      GOTO 30
   40 CONTINUE
      WRITE(*,'(A,I0)') 'dot=',DOT
      T=U(31)
      WRITE(*,'(A,I0)') 'u31=',T
      T=W(0)
      WRITE(*,'(A,I0)') 'w0=',T
      END

File: stress/t/arr_dotprod.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t i,n,a,b,dot,t;
  int64_t U[32],W[32];
  n=32;
  for(i=0;i<n;i++){ U[i]=i-16; W[i]=2*i+3; }
  dot=0;
  for(i=0;i<n;i++){ a=U[i]; b=W[i]; dot=dot+a*b; }
  printf("dot="); printf("%ld",(long)dot); printf("\n");
  t=U[31]; printf("u31="); printf("%ld",(long)t); printf("\n");
  t=W[0]; printf("w0="); printf("%ld",(long)t); printf("\n");
  return 0;
}
